### PR TITLE
Skip Codecov.submit_local test if not in a git repo; Use LibGit2.GitRepoExt to (try to) load git repository

### LINKED ISSUE #159: Bitbucket Pipelines Codecov submission no longer works
I previously used Coverage.jl in Bitbucket pipelines without any issues. With the newest update to LibGit2, I am getting an error. I have a private Docker container with all my dependencies. The build script looks like:

```
# Bitbucket portions
umask 000
git clone --branch="master" --depth 50 https://x-token-auth:$REPOSITORY_OAUTH_ACCESS_TOKEN@bitbucket.org/MYREPO.jl.git $BUILD_DIR ; git reset --hard ### ; git remote set-url origin git@bitbucket.org:MYREPO.git
chmod 777 $BUILD_DIR
# my commands, worked previously
mkdir ~/.julia/v0.6/MYPKG && cp -r ./ ~/.julia/v0.6/MYPKG
julia -e 'Pkg.clone(pwd()); Pkg.build("MYPKG"); Pkg.test("MYPKG"; coverage=true)'
julia -e 'cd(Pkg.dir("MYPKG")); using Coverage; Codecov.submit_local(Codecov.process_folder())'
```
The testing and file coverage generation goes fine. Where previously it would submit the report, I now get:
```
ERROR: GitError(Code:ENOTFOUND, Class:Repository, Could not find repository from '/root/.julia/v0.6/MYPKG/..')
Stacktrace:
 [1] macro expansion at ./libgit2/error.jl:99 [inlined]
 [2] Base.LibGit2.GitRepo(::String) at ./libgit2/repository.jl:10
 [3] #submit_local#11(::Array{Any,1}, ::Function, ::Array{Coverage.FileCoverage,1}, ::String) at /root/.julia/v0.6/Coverage/src/codecovio.jl:170
 [4] submit_local(::Array{Coverage.FileCoverage,1}) at /root/.julia/v0.6/Coverage/src/codecovio.jl:170
```

I will investigate this and update if I can find a way to set up the build script in a way that works. Any hints or help is appreciated.

### PR DESCRIPTION
**Pull request by fingolfin**
This should fix the test failures reported by JuliaCIBot on https://github.com/JuliaLang/METADATA.jl/pull/19092

Also use `LibGit2.GitRepoExt` instead of `LibGit2.GitRepo` to (try to) load git repository. This should fix #159 and render #160 obsolete.

### CODE REVIEW

File: src/codecovio.jl [hunk 1]
```diff
@@ -158,8 +158,8 @@ module Codecov
     a local git installation, rooted at `dir`. A repository token should be specified by a
     `token` keyword argument or the `CODECOV_TOKEN` environment variable.
     """
-    function submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=joinpath(pwd(),".."); kwargs...)
-        LibGit2.with(LibGit2.GitRepo(dir)) do repo
+    function submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=pwd(); kwargs...)
+        LibGit2.with(LibGit2.GitRepoExt(dir)) do repo
```
Comment (ararslan) [inline src/codecovio.jl]: Why `GitRepoExt`? I'm not very familiar with the difference. Does this create a git repo at the given directory if one doesn't already exist or something?

Comment (fingolfin) [reply src/codecovio.jl]: No, it works in subdirectories. So if dir does not point a git repo, but just as a subdir of a git repo, GitRepoExt walks back the path hierarchy until it finds the root of the git repo.

This means that `GitRepoExt(pwd())` will work both when called from the root dir of the repo, and also when called from inside the test/ dir.

Anyway, this PR is not yet complete; I plan to resolve #159 and #160 with it, and perhaps some more.

Comment (ararslan) [reply src/codecovio.jl]: Okay, thanks.

Comment (fingolfin) [reply src/codecovio.jl]: PR is now complete from my POV (i.e., modulo review change requests)

**Comment by ararslan** [comment]
Looks okay to me if CI agrees.

**Comment by fingolfin** [comment]
Weird, Travis tests completed, but here they are still shown as "Pending".

**Pull request merged**